ZeroWasteFriendly/Helpers/MainPresenter.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  MainPresenter.swift
+//  ZeroWasteFriendly
+//
+//  Created by Marek Skrzelowski on 08/02/2020.
+//  Copyright © 2020 vandermesis. All rights reserved.
+//
+
+import UIKit
+
+protocol SpinnerPresenter {
+    func toggleSpinner(_ state: Bool)
+}
+
+protocol AlertPresenter {
+    func presentAlert(title: String, message: String)
+    func presentError(_ error: Error)
+}
+
+class MainPresenter<T: SpinnerPresentable & AlertPresentable>: SpinnerPresenter, AlertPresenter {
+
+    weak var controller: T?
+
+    func toggleSpinner(_ state: Bool) {
+        controller?.toogleSpinner(state)
+    }
+
+    func presentAlert(title: String, message: String) {
+        controller?.presentAlert(title: title, message: message)
+    }
+
+    func presentError(_ error: Error) {
+        controller?.presentAlert(title: "Error", message: error.localizedDescription)
```

Comment:
You can also make that Error extension to display userfriendlymessages and have it in project